src/clj/game/cards-identities.clj

```diff
@@ -114,6 +130,20 @@
                               :msg (msg "draw a card")
                               :effect (effect (draw 1))}}}
 
+   "Fringe Applications: Tomorrow, Today"
+   {:effect draft-points-target
+    :events (let [fap {:req (req (let [facs (frequencies (map :faction (filter :rezzed (all-installed state :corp))))
+                                       hb (get facs "Haas-Bioroid" 0)
+                                       ji (get facs "Jinteki" 0)
+                                       nb (get facs "NBN" 0)
+                                       we (get facs "Weyland Consortium" 0)]
+                                   (and (> we ji) (> we hb) (> we nb))))
+                       :msg (msg "place 1 advancement token on " (card-str state target))
+                       :prompt "Choose a piece of ICE to place 1 advancement token on it" :player :corp
```

Comment:
If we are pedantic this should be "advancement counter" as per the card. But they are the same thing as ruled by Lucas.